src/widgetastic_patternfly/__init__.py

```diff
@@ -227,39 +230,58 @@ def dismiss(self):
         for message in self.messages:
             message.dismiss()
 
-    def assert_no_error(self):
-        self.logger.info('asserting there are no error messages')
-        for message in self.messages:
-            if message.type not in {'success', 'info', 'warning'}:
-                self.logger.error('%s: %r', message.type, message.text)
-                raise AssertionError('assert_no_error: {}: {}'.format(message.type, message.text))
-            else:
-                self.logger.info('%s: %r', message.type, message.text)
-
-    def assert_message(self, text, t=None, partial=False):
-        log_part = 'partial match of' if partial else ''
+    def match_messages(self, text=None, t=None, partial=False, inverse=False):
+        msg_type = t if isinstance(t, (tuple, list, set, type(None))) else (t, )
+        op = bool if inverse else not_
```

Comment:
Discussed an in-line comment here, because I actually read this as exactly the opposite of what it does, and then was confused by the logic in the blocks below.

Its the right choice of implementation, just needs some comments to clarify.